src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -55,15 +66,24 @@ def run_escalating_forecast() -> None:
     default_retry_delay=60,
 )
 @retry
-def generate_forecasts_for_projects(project_ids: List[int]) -> None:
+def generate_forecasts_for_projects(project_ids: List[int], forecast_for_ongoing=False) -> None:
+    query_for_archived_until_escalating = Q(
+        status=GroupStatus.IGNORED, substatus=GroupSubStatus.UNTIL_ESCALATING
+    )
+    query_for_ongoing = Q(status=GroupStatus.UNRESOLVED, substatus=GroupSubStatus.ONGOING)
+
+    combined_query = (
+        (query_for_archived_until_escalating | query_for_ongoing)
+        if forecast_for_ongoing
+        else query_for_archived_until_escalating
+    )
+
     for until_escalating_groups in chunked(
         RangeQuerySetWrapper(
             Group.objects.filter(
```

Comment:
I'm thinking out loud what could be ways to prevent calling `generate_forecasts_for_projects` twice.

```
filter_groups = Q(
    status=GroupStatus.IGNORED, substatus=GroupSubStatus.UNTIL_ESCALATING
)
if forecast_for_ongoing:
    filter_groups = # Query until escalating groups AND ongoing
    
for until_escalating_groups in chunked(
        RangeQuerySetWrapper(
            Group.objects.filter(
                project_id__in=project_ids,
                last_seen__gte=datetime.now() - timedelta(days=7),
            ).filter(filter_groups),
            step=ITERATOR_CHUNK,
        ),
        ITERATOR_CHUNK,
    ):
        generate_and_save_forecasts(groups=until_escalating_groups)
```


or 
```
for until_escalating_groups in chunked(
        # Not using filter()
        RangeQuerySetWrapper(
            Group.objects.filter(
                project_id__in=project_ids,
                last_seen__gte=datetime.now() - timedelta(days=7),
            ),
            step=ITERATOR_CHUNK,
        ),
        ITERATOR_CHUNK,
    ):
        filtered_out_groups = # Iterate over until_escalating_groups and use a function that can exclude groups based on forecast_for_ongoing
        generate_and_save_forecasts(groups=until_escalating_groups)
```